### ISSUE TITLE
b64 InvalidCastException

### ISSUE THREAD

**Issue by fdres** [dvsekhvalnov/jose-jwt#121]
Hi.
Inside private static byte[] DecodeBytes(...) in JWT.cs there is this piece of code:
`object value;
if (headerData.TryGetValue("b64", out value))
{
     b64 = (bool) value;
}`

If value is "false" or "true" this will throw an InvalidCastException.
Would you consider the change to 
`b64 = Convert.ToBoolean(value);`

**Comment by dvsekhvalnov** [comment]
Sure can absolutely do it. 

Are you facing this issue in real world? Mind unit test or sample data?

I also happy to accept PR :)

**Comment by fdres** [comment]
Thanks for your response.
I'm using the nuget in an ASP.NET Core API for decoding.
I use postman as a client with javascript pre-request script to sign the request body

Client Code Sample
`var oHeader = {
	"alg": "PS256", 
	"typ": "JOSE",
	"cty": "application/json",
	"b64": "true"
};
var sHeader = JSON.stringify(oHeader);
var token = KJUR.jws.JWS.sign("PS256", sHeader, sPayload, privateKey);`

**Comment by dvsekhvalnov** [comment]
@fdres thanks for example.

May i ask why you putting string value for b64 header instead of boolean?

RFC7797 pretty clear defines (https://tools.ietf.org/html/rfc7797#section-3): 
```
The "b64" value is a JSON boolean, with a default value of "true".
```

Obviously i don't mind to add fix to support string values as well if it helps in real world. Just trying to understand what is real use case for it?

**Comment by fdres** [comment]
I started using the nuget in version 2.4.0. When I sent the b64 header as boolean I got an IntegrityException with the message "Invalid Signature". When I changed the value to a string it validated the payload.
I don't know if the problem with the boolean value is with the js framework I'm using to sign the request ( I am not a javascript developer and I was searching for something to use with postman).

**Comment by dvsekhvalnov** [comment]
I see. Are you using https://kjur.github.io/jsrsasign/ ? 

Probably i need to craft some example.

**Comment by fdres** [comment]
_"Are you using https://kjur.github.io/jsrsasign/ ?"_
Yes

**Comment by dvsekhvalnov** [comment]
@fdres i'm not super sure but i can't find that `jsrsasign` support unencoded payload.
It seems to match what you are observing:
 - when you provide boolean `b64` value, jsrsasign not handling it and still applies base64 encoding for payload. On verification part if it obviously fails as jose-jwt expect unencoded payload

Pretty simple to verify, just generate 2 tokens via jsrsasign: one with b64=true, another with b64=false and post here or compare yourself. Pretty sure you'll get identical payload part within.

**Comment by fdres** [comment]
@dvsekhvalnov 
That's true. `jsrasign` does not support unencoded payload (yet?).
My original issue was that I sent a b64 header with value "true" and it couldn't be parsed. That caused my api to throw a 500 http status code.
The problem I was having with jose-jwt and lead me to send string value instead of boolean does not seem to occur any more (probably I was misusing the library).
Anyway. I think is a good practice to either try parse any value sent at the header, or throw a proper exception.
Thanks for your time

**Comment by dvsekhvalnov** [comment]
Yeah, for sure i agree we should improve error handling, InvalidCastException is terrible )

I'm slightly disagree with accepting "string" values for b64 header as it against spec and no real use case. 

Will publish fix in next minor version.

**Comment by ThaDaVos** [comment]
Running into the same exception - but in my case it seems `b64` is missing - I needed to create my own `JsonMapper` for `System.Text.Json` due to using a custom object as result and the need to use the `SourceGenerator` for `System.Text.Json` as my application is NativeAoT.

At least I think `b64` is missing as when i use https://www.jwt.io/ to decode it, it's missing from the header

**Comment by ThaDaVos** [comment]
I was apparently missing the special Converter for `object` - here is mine below:
```csharp
    class ObjectToInferredTypesConverter : JsonConverter<object>
    {
        public override object? Read(
            ref Utf8JsonReader reader,
            Type typeToConvert,
            JsonSerializerOptions options) => reader.TokenType switch
            {
                JsonTokenType.True => true,
                JsonTokenType.False => false,
                JsonTokenType.Number when reader.TryGetInt64(out long l) => l,
                JsonTokenType.Number => reader.GetDouble(),
                JsonTokenType.String when reader.TryGetDateTime(out DateTime datetime) => datetime,
                JsonTokenType.String => reader.GetString()!,
                JsonTokenType.Null => null,
#pragma warning disable IL2026 // Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code
#pragma warning disable IL3050 // Calling members annotated with 'RequiresDynamicCodeAttribute' may break functionality when AOT compiling.
                JsonTokenType.StartObject => JsonSerializer.Deserialize(ref reader, typeof(IDictionary<string, object>), options),
                JsonTokenType.StartArray => JsonSerializer.Deserialize(ref reader, typeof(IEnumerable<object>), options),
#pragma warning restore IL3050 // Calling members annotated with 'RequiresDynamicCodeAttribute' may break functionality when AOT compiling.
#pragma warning restore IL2026 // Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code
                _ => JsonDocument.ParseValue(ref reader).RootElement.Clone()
            };

        public override void Write(
            Utf8JsonWriter writer,
            object objectToWrite,
            JsonSerializerOptions options)
        {
#pragma warning disable IL2026 // Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code
#pragma warning disable IL3050 // Calling members annotated with 'RequiresDynamicCodeAttribute' may break functionality when AOT compiling.
            JsonSerializer.Serialize(writer, objectToWrite, typeof(object), options);
#pragma warning restore IL3050 // Calling members annotated with 'RequiresDynamicCodeAttribute' may break functionality when AOT compiling.
#pragma warning restore IL2026 // Members annotated with 'RequiresUnreferencedCodeAttribute' require dynamic access otherwise can break functionality when trimming application code
        }
    }
```
